src/pages/spawn/spawnHost/spawnHostModal/Widgets/DistroDropdown.tsx

```diff
@@ -0,0 +1,129 @@
+import styled from "@emotion/styled";
+import { uiColors } from "@leafygreen-ui/palette";
+import { Overline } from "@leafygreen-ui/typography";
+import SearchableDropdown, {
+  SearchableDropdownProps,
+} from "components/SearchableDropdown";
+import ElementWrapper from "components/SpruceForm/ElementWrapper";
+import { SpruceWidgetProps } from "components/SpruceForm/Widgets/types";
+import { size } from "constants/tokens";
+
+const { gray } = uiColors;
+
+interface DistroValue {
+  value: string;
+  isVirtualWorkstation: boolean;
+}
+interface OptionValue {
+  title: string;
+  distros: DistroValue[];
+}
+
+interface DistroEnum {
+  options: {
+    enumOptions: Array<{
+      schema: {
+        isVirtualWorkstation: boolean;
+      };
+      label: string;
+      value: string;
+    }>;
+  };
+}
+
+export const DistroDropdown: React.VFC<
+  DistroEnum &
+    SpruceWidgetProps & {
+      options: Pick<SearchableDropdownProps<string>, "data-cy">;
+    }
+> = ({ options, label, onChange, ...rest }) => {
+  const {
+    "data-cy": dataCy,
+    ariaLabelledBy,
+    elementWrapperCSS,
+    enumOptions,
+  } = options;
+
+  const searchableOptions = enumOptions.reduce<OptionValue[]>(
+    (accum, { schema, value }) => {
+      const { isVirtualWorkstation } = schema;
+      accum[isVirtualWorkstation ? 0 : 1].distros.push({
```

Comment:
I understand what this is doing now but do you think we could make it more verbose or add some comments so its more clear in the future when we come back to this.